### ISSUE TITLE
Unable to install

### ISSUE THREAD

**Issue by Kinpzz** [Andrew-Qibin/SalMetric#2]
Hi, Andrew

I have trouble with both C++ and python installing method. They were all failed, and I can get little information from the debug log. Should I modify the program?

The log are as following:
* C++:
![](http://ww1.sinaimg.cn/mw690/6177e8b1gy1fl74276ovuj20yw0nhaj5.jpg)
* Python:
![](http://ww1.sinaimg.cn/mw690/6177e8b1gy1fl744k8uduj20z40qs45z.jpg)

**Comment by houqb** [comment]
Yeah, sorry for that. At present, you may need to modify the code. I will fix this later on.

**Comment by Kinpzz** [comment]
Thanks for your reply. It is very kind of you. However I don't know where I should modify because I barely get anything from the logs information. And I don't have any error information about OpenCV2.

**Comment by houqb** [comment]
Some of the member functions of OpenCV2 are different from opencv3. You can try to compile the C++ part with gcc.

**Comment by Sucran** [comment]
@Kinpzz I had the same problem and I solved it by
1. make sure opencv3 installed correctly (followed opencv tutorial)
2. modified the setup.py for the right path for you envs
3. using the command line "python setup.py install"
4. cp ./build/lib.linux-x86_64-2.7/build/salmetric.so /home/user/anaconda2/envs/xxx/lib/python2.7/site-packages/salmetric.so
5. then you can find the salmetric module
if you want to modified the salmetric.cpp file, you can modified it and replace the file  ./src/salmetric.cpp, then follow the 4 and 5 step

@Andrew-Qibin it is very kind of you to open this code. Can you give me some advice if I want to modify the code to return MaxFb and mae value as a list in the member function do_evaluation() ?

**Comment by yuanyao366** [comment]
@Kinpzz I have the trouble with both C++ and python installing method. Have you sloved it?
@Sucran I have opencv 2.4 installed in my minicoda. Then I install opencv 3.2 and add the command 
set(OpenCV_DIR "/usr/local/opencv320/share/OpenCV")  to the CMakeLists.txt. And I follow the tip3,4,5 as you suggested. But I still meet such problem:
 
>>> import salmetric
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /root/Util/miniconda/lib/python2.7/site-packages/salmetric.so: undefined symbol: _ZN2cv6imreadERKSsi

Have you some solutions?
And I am not particularly clear about the tip 2. Can you describe it more clearly?
Thanks a lot!

**Comment by sofmonk** [comment]
> @Kinpzz I had the same problem and I solved it by
> 
> 1. make sure opencv3 installed correctly (followed opencv tutorial)
> 2. modified the setup.py for the right path for you envs
> 3. using the command line "python setup.py install"
> 4. cp ./build/lib.linux-x86_64-2.7/build/salmetric.so /home/user/anaconda2/envs/xxx/lib/python2.7/site-packages/salmetric.so
> 5. then you can find the salmetric module
>    if you want to modified the salmetric.cpp file, you can modified it and replace the file  ./src/salmetric.cpp, then follow the 4 and 5 step
> 
> @Andrew-Qibin it is very kind of you to open this code. Can you give me some advice if I want to modify the code to return MaxFb and mae value as a list in the member function do_evaluation() ?

Hi, I still get the error ` AttributeError: module 'salmetric' has no attribute 'do_evalution'` after correctly importing salmetric module.

**Comment by lightas** [comment]
> > @Kinpzz I had the same problem and I solved it by
> > 
> > 1. make sure opencv3 installed correctly (followed opencv tutorial)
> > 2. modified the setup.py for the right path for you envs
> > 3. using the command line "python setup.py install"
> > 4. cp ./build/lib.linux-x86_64-2.7/build/salmetric.so /home/user/anaconda2/envs/xxx/lib/python2.7/site-packages/salmetric.so
> > 5. then you can find the salmetric module
> >    if you want to modified the salmetric.cpp file, you can modified it and replace the file  ./src/salmetric.cpp, then follow the 4 and 5 step
> > 
> > @Andrew-Qibin it is very kind of you to open this code. Can you give me some advice if I want to modify the code to return MaxFb and mae value as a list in the member function do_evaluation() ?
> 
> Hi, I still get the error ` AttributeError: module 'salmetric' has no attribute 'do_evalution'` after correctly importing salmetric module.

Same. Any solutions?

**Comment by uvasudha** [comment]
> > > @Kinpzz I had the same problem and I solved it by
> > > 
> > > 1. make sure opencv3 installed correctly (followed opencv tutorial)
> > > 2. modified the setup.py for the right path for you envs
> > > 3. using the command line "python setup.py install"
> > > 4. cp ./build/lib.linux-x86_64-2.7/build/salmetric.so /home/user/anaconda2/envs/xxx/lib/python2.7/site-packages/salmetric.so
> > > 5. then you can find the salmetric module
> > >    if you want to modified the salmetric.cpp file, you can modified it and replace the file  ./src/salmetric.cpp, then follow the 4 and 5 step
> > > 
> > > @Andrew-Qibin it is very kind of you to open this code. Can you give me some advice if I want to modify the code to return MaxFb and mae value as a list in the member function do_evaluation() ?
> > 
> > 
> > Hi, I still get the error ` AttributeError: module 'salmetric' has no attribute 'do_evalution'` after correctly importing salmetric module.
> 
> Same. Any solutions?

have you solved the issue, unable to figure out the solution

**Comment by uvasudha** [comment]
> @Kinpzz I had the same problem and I solved it by
> 
> 1. make sure opencv3 installed correctly (followed opencv tutorial)
> 2. modified the setup.py for the right path for you envs
> 3. using the command line "python setup.py install"
> 4. cp ./build/lib.linux-x86_64-2.7/build/salmetric.so /home/user/anaconda2/envs/xxx/lib/python2.7/site-packages/salmetric.so
> 5. then you can find the salmetric module
>    if you want to modified the salmetric.cpp file, you can modified it and replace the file  ./src/salmetric.cpp, then follow the 4 and 5 step
> 
> @Andrew-Qibin it is very kind of you to open this code. Can you give me some advice if I want to modify the code to return MaxFb and mae value as a list in the member function do_evaluation() ?

@Sucran Did you had any error while calling do_evaluation(), I am getting an error saying salmetric has no atrribute do_evaluation().
Any suggestion will be helpful, thank you.

**Comment by hit421** [comment]
do_evalution() should be do_evaluation()
